\documentclass[12pt]{article}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%% Setup %%%%%%%%%%%%%%%%%%%%%%%%%
\usepackage[margin=1in]{geometry}
\usepackage{fmtcount} % 1st, 2nd...
\usepackage[english]{babel}
\usepackage{soul} % spacing
\usepackage{enumerate} % change enum label
\usepackage{framed} % frame pagebreak
\usepackage{authblk}
\usepackage{setspace}
\usepackage{import}

%%%%% Math, Code 
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{amsthm}
\usepackage{amscd}
\usepackage{array}
\usepackage{mathtools} 
\usepackage{verbatim}
\usepackage{fancyvrb} % verbatim
\usepackage[linesnumbered]{algorithm2e}

%%%%% Graphic, Figure, Table
\usepackage{graphicx}
\usepackage{epstopdf}
\usepackage{xcolor}
\usepackage{caption}
\usepackage{subcaption}
\usepackage{multirow} % table
\usepackage{longtable} % table to next page
\usepackage{rotating}
\usepackage{wrapfig} % wrap fig, tbl with text

%%%%% Bib, Cite
\usepackage{url}
\usepackage{cite}
\usepackage{fancyref}
\usepackage{hyperref}
\hypersetup{linktocpage}

%%%%%% Setup
\captionsetup{compatibility=false} %subfig
\hypersetup{colorlinks=false}
\setlength{\parskip}{0.5em}

\newtheorem{thm}{Theorem}[section]
\newtheorem{cor}[thm]{Corollary}
\newtheorem{lem}[thm]{Lemma}
\newtheorem{prop}[thm]{Proposition}
\newtheorem{dfn}{Definition}
\newtheorem{ex}{Example}
\newtheorem{cmt}{Cmt}[section]
\newtheorem{rmk}{Remark}[section]
\newcommand{\rb}[1]{\raisebox{-.5em}[0pt]{#1}}
\newcommand{\1}{{\rm 1}\kern-0.24em{\rm I}}
\newcommand{\cN}{\mathcal{N}}
\renewcommand{\mid}{\, | \ , }
\renewcommand\Affilfont{\normalsize}
% \renewcommand{\baselinestretch}{1.3}

\DeclareMathOperator*{\argmin}{arg\,min}
\DeclareMathOperator*{\argmax}{arg\,max}
\DeclareMathOperator*{\bbr}{\mathbb{R}}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Multidimensional scaling informed by $F$-ratio: Visualizing microbiome for inference}
\author[1]{Hyungseok Kim$^*$}
\author[2]{Soobin Kim$^*$}
\author[3]{Megan M. Morris}
\author[3]{Jeffrey A. Kimbrel}
\author[3]{Xavier Mayali}
\author[1]{Cullen R. Buie$^\dagger$}
\affil[1]{Massachusetts Institute of Technology}
\affil[2]{University of California, Davis}
\affil[3]{Lawrence Livermore National Laboratory}
\date{}
\begin{document}


\maketitle
\def\thefootnote{*}\footnotetext{Equal contribution.}
\def\thefootnote{$\dagger$}\footnotetext{Correspondence. {\tt crb@mit.edu}}
\def\thefootnote{1}\footnotetext{Code available at \url{https://github.com/hyu-kim/mds-hypothesis-testing}.}

\begin{abstract}
    Multidimensional scaling (MDS) is a dimension reduction technique that preserves pairwise distances between observations and has commonly been used for displaying multivariate biological data.  
    Recent MDS approaches have been based on supervised learning where label information is incorporated for training.  
    However, their configurations heavily depend on the choice of hyperparameters, making them susceptible to false positives.  
    In this study, we present a weakly supervised MDS approach informed by dispersions of observations that share a common binary label ($F$-ratio).  
    We compare our method with existing dimension reduction approaches and show that our visualization accurately represents the $F$-ratio while consistently preserving the global structure.  
    Using a host-associated microbiome dataset, we show that this new method better illustrates the community's response to the host, suggesting its potential impact on microbiology and ecology data analysis \textsuperscript{1}. \newline

    \noindent \textbf{Keywords}: Multidimensional scaling, microbiome, \textit{F}-statistic, dimensionality reduction.
\end{abstract}


%%%%
\section{Introduction}
% Visualizing multivarate dataset is a key process in interpreting microbial community data
Technical advances in recent microbiome research have led to an increase in the size and dimensionality of biological data and empowered the use of data analysis tools.
% a demand in using appropriate visualization tools for interpreting the data. 
% in the past several decades ...
% For example, the cost of sequencing bacterial samples have been lowered by \textcolor{red}{XX times for the last Y years}, enabling a amore accessible quantification of microbial community compositions. 
Analysis of the microbiome data is performed in two ways.
First, the multivariate structure is interpreted by visualizing it at a lower dimensional space (two- or three-dimensional), a process termed as the dimensionality reduction.
Second, a quantitative significance is inferred on the multivariate data by performing a hypothesis testing or a statistical regression. 

Microbiome samples present a unique structure among multivariate data in that they are sparse, compositional \textcolor{red}{[refs]}, and analyzed in a biological context based on taxonomical/phylogenetic tree \textcolor{red}{refs}.
Multidimensionals caling (MDS) has been a good visualzation approach for such data type due to its purpose of preserving pairwise distance between the community samples.

By extracting an essential information from the biological data, the dimensionality reduction seeks a visual representation of the multivariate in a lower dimensional space. 
To retain a consistent data structure while performing the dimension reduction, a configuration is sought in a way that preserves dissimilarity between samples, a process called as the multidimensional scaling (MDS). 
Compared to other nonlinear methods in dimensionality reduction \cite{tenenbaum00, mcInnes18, maaten08}, MDS is known to retain a global structure and represent a long-range interaction between samples, allowing its popularity for a long time since its inception \cite{Demaine21}. 

In the classical MDS, the configuration is determined in a way that minimizes the difference between the dissimilarity in the original and the lower dimensional space.
Because biological observations often present a compositional structure (highly skewed, zero-inflated \cite{gijbels13}), 
choosing an appropriate dissimilarity metric is an important step for processing the data and representing in the low dimensional space.
% In microbial ecology where the observational data are compositional (i.e., a sample is comprised of different species expressing a level of abundance), the dissimilarity is measured by a difference in expression level of species within samples. 
For example, when interpreting a microbial community dataset such as 16S rRNA gene expression, a distance metric such as Unifrac \cite{lozupone05, lozupone07} allows to incorporate taxonomic information as well as the compositional structure. 

Whereas the dimensionality reduction provides a qualitative interpretation on the multivariate data structure, statistical inference such as hypothesis testing is carried out at the same time to give a quantitative analysis on the structural difference by biological responses. 
% For most of the biological data, a probabilistic distribution cannot be assumed a priori, and a nonparametric hypothesis testing can be constructed instead. 
% When the response is assumed to be independent between the samples, the statistic can be readily obtained by permuting the data labels \cite{holmes96}. 
One way to test the difference between response groups is to compare dispersions of the multivariate structure that are calculated across or within the sample groups, which gives an \textit{F}-statistic\cite{gijbels13}.
The process employs a different information than those used for MDS, in that it is unsupervised learning method and does not use label information. 
Therefore, MDS configuration under a low dimension (e.g., 2) may not fully explain a small but statistically meaningful difference between groups of different treatment.
% This is because most MDS (e.g., PCA, PCoA) are unsupervised learning, whereas the hypothesis testing aims to infer whether responses (e.g., class, label) can influence the distribution of each group of observations. 
% For example, a configuration from the classical MDS is not able to explain a small but statistically meaningful difference between groups of different treatment. 

The insufficient explanation by the classical MDS encourages to revise the method and to address a structural hypothesis by including an external information which is conferred by the responses or class labels \cite{ding18}. 
Broadly termed as the confirmatory MDS, this approach is based on applying an external constraint to the classical MDS, thereby providing a more contextual illustration of data structure \cite{borg97b}.
Given that several local configurations can be produced from classical MDS, there is a notion accepting altered configurations up to a point where they do not deviate too much from the original configuration. 

In these confirmatory MDS methods, an objective function is constructed by adding a confirmatory term to the stress, weighed by a hyperparameter, and informed by the labels or responses. 
These confirmatory MDS methods have successfully visualized the multivariate structure in a way that differentiates sample groups with a discriminative purpose \cite{cox93, yang18}.
However, choosing a proper hyperparameter remains as a bottleneck towards a broader application of these inventive methods. 
For example, setting a high hyperparameter can result in an undesirable stress and a misleading configuration heavily distorted from the original.

In this work we introduce a weakly supervised version of MDS informed by \textit{F}-statistic of multivariate dataset, which we refer to as FMDS.
In other words, FMDS is informed by a hypothesis testing results under a binary class setting using $F$-statistic. 
The proposed approach is motivated by a purpose to explain a statistical difference between groups, if any, and to incorporate it to the confirmatory analysis by combining with classical MDS configuration. 
Our motivation distinguishes itself from existing variations in confirmatory MDS because the method does not target to directly discriminate between groups.
\textcolor{blue}{This allows FMDS to avoid a false positive that may arise when a high hyperparameter is imposed to a classification/superivsing term when configuring  binary-labeled dataset onto a two-dimensional space. }
Furthermore, by characterizing the proposed framework we show that FMDS configuration is less dependent to the choice of the model hyperparameter, mediating the existing issue with the distortion.


%%%%
\section{Related Work}

\paragraph{Classical multidimensional scaling:}
Consider a balanced design where the number of total observations is $N$, and each observation $x_i$ is $S$-dimensional, pertaining to a set of labels $y_i\in\{0,1\}$ for every $i=1,\cdots N$. 
Using on the set of observations $(x_1,\cdots x_N)$, a pairwise distance $d_{ij}\in\{0,\infty\}$ between $x_i$ and $x_j$ can be obtained using an appropriate distance metric (e.g., Euclidean, Bray-Curtis \cite{bray57}, UniFrac \cite{lozupone05}).
In the classical MDS, a lower-dimensional configuration $\mathbf{z} = (\mathbf{z}_1, \cdots \mathbf{z}_N)\in \bbr^{N \times 2}$ is sought in a way that preserves the pairwise distance while the dimension is reduced. 
This is enabled by minimizing the ``raw stress'' \cite{borg05c}: ${\rm Stress} = \frac{1}{2}\sum_{i,j} (d_{ij} - \| \mathbf{z}_i - \mathbf{z}_j \|_2 )^2$.
% a difference in the pairwise distance between the dimensions and summing the difference for all pairs, so-called 
Therefore, the classical MDS is unsupervised learning and it does not require a set of labels $y_i$ for training.

\paragraph{Supervised multidimensional scaling \cite{witten11}:}
Supervised multidimensional scaling imposes an additional constraint to the configuration using a confirmatory feature that is informed by class labels. 
The purpose is to discriminate observations by the labels as well as carrying out the task of minimizing the stress.
To achieve this an objective function $O(\mathbf{z})$ is proposed by summing the raw stress and a confirmatory term, 
\begin{equation}
O(\mathbf{z}) = \frac{1}{2}(1-\alpha)\sum_{i,j} (d_{ij} - \| \mathbf{z}_i - \mathbf{z}_j \|_2 )^2 + \alpha\sum_{i,j:y_{j}>y_{i}} (y_j - y_i) \sum_{s=1}^2 \left(\frac{D_{ij}}{\sqrt{2}}-(z_{js} - z_{is})\right)^2,
\end{equation}
where the confirmatory term contains labels $y_i$ so that the configuration points $\mathbf{z}_i$ and $\mathbf{z}_j$ are not too far apart when $y_i\approx y_j$.
In addition, a hyperparameter $\alpha$ is chosen to determine the degree of classification task over the stress minimization.

\paragraph{Non-parametric multivariate analysis of variance:}
When testing a difference between groups of multivariate data, test statistics are obtained through calculating variances or a sum of distances / dissimilarities. 
Most widely used metric is $F$-statistic, a ratio between two sums respectively calculated from inter- and intra-group variances.
However, performing $F$-test necessitates an assumption that the observations follow a normal distribution with common variance, which is hardly met in practice when dealing with biological dataset.
In this non-parametric context, an alternative approach has been proposed by permuting the labels iteratively to create an empirical distribution \cite{anderson01, holmes96}.
% Using the distribution and (pseudo) $F-$statistic, a statistical significance can be obtained \cite{anderson01}. 
First, define the pseudo $F$-ratio as
\begin{equation} \label{eq:pseudof}
F = \frac{\sum_{i,j}d_{ij}^2 - 2\sum_{i,j}\1\{y_i=y_j\}d_{ij}^2}{2\sum_{i,j}\1\{y_i=y_j\}d_{ij}^2}\cdot (N-2),
\end{equation}
where $\1\{\cdot\}$ denotes an indicator function. 
While the pseudo $F$ does not always follow the $F$-distribution under the non-parametric setting, an empirical distribution can be constructed instead by `permuting' the labels for a large enough size of dataset \cite{gijbels13, anderson01}.
In other words, denote $F^\pi$ as a new $F$-ratio that is obtained from the permuted labels, and by iterating the permutation to obtain a $P$-value,
\begin{equation}
P = \frac{\textrm{Number of case where }(F^\pi\geq F)}{\textrm{Number of total repeat}}.
\end{equation}
The procedure was first proposed by Anderson \cite{anderson01}, known as the permutational multivariate analysis of variance (PERMANOVA).


%%%%
\section{Approach}
We have set our goal to incorporate the multivariate hypothesis testing result for determining an MDS configuration of the dataset.
To achieve the goal, we take a self-supervised approach with the following two steps.
First, a two-dimensional configuration is trained by performing the classical MDS (i.e., unsupervised).
Next, the configuration points are adjusted in a way that best represents a $P$ value calculated from the multivariate, binary-labeled dataset.
% we propose a new approach called the $F$-informed multidimensional scaling (FMDS). 
This is enabled by adding a confirmatory term to the raw stress, a similar approach proposed by the supervised MDS \cite{witten11},
\begin{equation}\label{eq:fmds}
    O(\mathbf{z}) = \underbrace{\frac{1}{2}\sum_{i,j} (d_{ij} - \| \mathbf{z}_i - \mathbf{z}_j \|_2 )^2}_\text{raw stress} + 
    \lambda\cdot \underbrace{\frac{1}{2} \left|\sum_{i,j} \left[1- 2\epsilon_{ij} \left(1+\frac{f_\mathbf{z}(F)}{N-2}\right)\right] \|\mathbf{z}_i - \mathbf{z}_j\|_2^2\right|}_\text{\textit{F}-informed confirmatory term},
\end{equation}
where $f_\mathbf{z}(F):\bbr\rightarrow\bbr$ is a scalar function that maps $F$-ratio from $S$-dimension onto the lower dimension, 2, which is determined by the configuration $\mathbf{z}$. 
Introducing the mapping $f_\mathbf{z}$ allows to address a difference in $F$-ratio distributions between $S$- and 2-dimension, 
thereby bridging a gap in the \textit{P} values.
A detailed derivation of $f_\mathbf{z}$ and a description on the confirmatory term is provided in Supplementary Note 1, and its pseudocode is shown in Algorithm \ref{alg:mapping}.

\RestyleAlgo{ruled}
\begin{algorithm}[h]
\SetKwProg{Fn}{Function}{:}{\KwRet $f_\mathbf{z}$}
\caption{Mapping from $F$ to $F_\mathbf{z}$ with random permutation.}
\label{alg:mapping}
\Fn{\textsc{mapping}{$(\mathbf{z}, d, y)$}}{
\KwIn{
\begin{tabular}{@{}ll}
    $\mathbf{z}$: & two-dimensional configuration \\
    $d$: & pairwise distance matrix \\
    $y$: & labels set
\end{tabular}
}
\KwOut{
\begin{tabular}{@{}ll}
    $f_\mathbf{z}$: & mapping function \\
    $L$: & sorted list of permuted $F$ \\
    $L_\mathbf{z}$: & sorted list of permuted $F_\mathbf{z}$
\end{tabular}
}
\For{$1\leq i \leq 999$}{
$y^\pi_1 \gets$ shuffle $y$ by a random permutation \\ 
$F^\pi \gets$ $F$-ratio by $d, y^\pi_1$ (Equation S1) \\
$L \gets$ append $F^\pi$ to $L$\\
$y^\pi_2 \gets$ shuffle $y$ by another random permutation \\
$F^\pi_\mathbf{z} \gets$ $F$-ratio by $\mathbf{z}, y^\pi_2$ (Equation S2) \\ 
$L_\mathbf{z} \gets$ append $F^\pi_\mathbf{z}$ to $L_\mathbf{z}$\\
}
$L \gets \text{sort}(L)$ \\
$L_\mathbf{z} \gets \text{sort}(L_\mathbf{z})$ \\
$f_\mathbf{z} \gets \text{LOESS}(L, L_\mathbf{z})$  
}
\end{algorithm}

\subsection{Majorize-Minimization (MM) algorithm}
Minimizing raw stress to perform MDS has been achieved by several iteration methods in the past few decades, part of which have originated in the field of graph drawing \cite{Kamada89}. 
Because the stress is a non-convex function in terms of $\mathbf{z}$, there is no global minimum and it may require a different initialization \cite{Demaine21, zheng19}.
This has also led with proposing several optimization methods including 2D Newton–Raphson \cite{Kamada89}, (stochastic \cite{zheng19}) gradient descent \cite{Kruskal64}, divide-and-conquer \cite{Yang06, Qu15}, and majorization \cite{deLeeuw88}. 

In our problem setting, we implemented the majorization (or Majorize-Minimization) approach to minimize the FMDS objective (Equation \ref{eq:fmds}).
% Because the confirmatory term in Equation \ref{eq:fmds} is quadratic in terms of $\mathbf{z}$, we can analytically derive the Majorize-Minimization (MM) algorithm to minimize $O(\mathbf{z})$ under a mild hyperparameter $\lambda$, a typical approach in MDS optimization task \cite{borg05c}. 
In brief, for every $k=1,\cdots N$, Equation \ref{eq:fmds} can be minimized by writing
\begingroup
% \allowdisplaybreaks
\begin{align}
\mathbf{z}_k^* 
&= \argmin_{\mathbf{z}_k}O(\mathbf{z}) \\
&= \argmin_{\mathbf{z}_k}\, \sum_{j=1}^N \left[1+\lambda\delta(\mathbf{z}) \left(1- 2\epsilon_{jk} \left(1+\frac{f_\mathbf{z}(F)}{N-2}\right)\right) \right] 
\|\mathbf{z}_k-\mathbf{z}_j\|_2^2 
- 2d_{jk}\|\mathbf{z}_k-\mathbf{z}_j\|_2
\label{eq:mm1}
\end{align}
\endgroup
where we define $\epsilon_{ij}$ and $\delta (\mathbf{z})$ as
\begin{align} \label{eq:simplify}
\epsilon_{ij} &= \1\{y_i = y_j\}, \quad
\delta (\mathbf{z}) = \text{sign}\left\{\,\sum_{i,j} \left[1- 2\epsilon_{ij} \left(1+\frac{f_\mathbf{z}(F)}{N-2}\right)\right] \|\mathbf{z}_i - \mathbf{z}_j\|_2^2\right\}.
\end{align}
Majorizing with Equation \ref{eq:mm1} gives a quadratic expression in terms of $\mathbf{z}$, as similarly described by Borg \cite{borg05c} and Witten \cite{witten11}.
Finally, a derivative is analytically taken with respect to $z_{ks}$ and is set to zero to find a local minimum, given that the expression is convex with assumptions. 
Detailed procedure is described in Supplementary note 2 and we provide the update rule in Algorithm \ref{alg:mm}. 

\RestyleAlgo{ruled}
\begin{algorithm}[h]
\SetKwProg{Fn}{Function}{:}{\KwRet $\mathbf{z}$}
\caption{Majorize-Minimization for FMDS}
\label{alg:mm}
\Fn{\textsc{mmfmds}{$(\lambda, d, y)$}}{
\KwIn{
\begin{tabular}{@{}ll}
    $\lambda$: & hyperparameter balancing raw stress and confirmatory term \\
    $d$: & pairwise distance matrix \\
    $y$: & labels
    
\end{tabular}
}
\KwOut{
\begin{tabular}{@{}ll}
    $\mathbf{z}$: & two-dimensional configuration
\end{tabular}
}
\For{$1\leq t \leq T$}{
\For{$1\leq k \leq N$}{
Apply Equation \ref{eq:pseudof} and compute $F$. \\
Apply Equation \ref{eq:simplify} and compute $\delta(\mathbf{z}).$ \\
Apply the \textsc{mapping} algorithm (\ref{alg:mapping}) and compute $f_\mathbf{z}(F)$. \\
\begin{align*}
& \mathbf{z}_k \gets
\frac{(N-2)}{N(N-2) - N\lambda\delta(\mathbf{z}) f_\mathbf{z}(F)}\times \\
&\qquad\left\{\sum_{j=1}^N\, \left[1+\lambda\delta(\mathbf{z}) \left(1- 2\epsilon_{jk} \left(1+\frac{f_\mathbf{z}(F)}{N-2}\right)\right) \right]  \mathbf{z}_{j} + d_{jk}\frac{\mathbf{z}_k -\mathbf{z}_j}{\|{\mathbf{z}}_k-\mathbf{z}_j\|_2}\right\}    
\end{align*}
}
}
}
\end{algorithm}


%%%%
\section{Experiments}
We first characterized the behavior of FMDS across a range of hyperparameters and compared its performance to benchmark dimensionality reduction methods, including supervised MDS \cite{witten11}, UMAP \cite{mcInnes18}, and a self-supervised neural network such as SimCLR \cite{chen20}.
For the evaluation we considered two types of datasets, simulated and experimental, both of which serve as examples where the two-dimensional configurations from classical MDS were not consistent with the hypothesis testing results.
Next, in Section \ref{sec:vis_fmds}, we demonstrate how FMDS can improve visualizing biological samples in a way that multivariate $F$ test results are addressed.

\subsection{Evaluating performance of \textit{F}-informed MDS}
We evaluated the performance of FMDS by quantifying a degree of deviation of the configuration $\mathbf{z}$ from the original distances $d_{ij}$.
It is carried out by calculating two metrics; the first is the ``normalized'' stress or Stress-1 \cite{borg05c},
\begin{equation}
\text{Stress-1} = \frac{\sum_{i,j} (d_{ij} - \| \mathbf{z}_i - \mathbf{z}_j \|_2)^2}{\sum_{i,j} d_{ij}^2},
\label{eq:mds_stress1}
\end{equation}
and the second using a Shepard diagram \cite{dexter18} and its correlation coefficient. 
% Both measures represent how much distortion has occurred in the configuration by each MDS.

\subsubsection{Simulated dataset} \label{sec:eval_sim}
We first considered a three-dimensional dataset of two balanced groups that follow normal distributions slightly different in means, $\mu_0$, $\mu_1$, but the same covariance matrix $D$. 
Such set of observations $x_i$ can be constructed as, for example, 
\begin{equation}
x_i \sim 
\begin{cases}
\cN \left(\mu_0, D\right), & i=1,2,\cdots 50 \\
\cN \left(\mu_1, D\right), & i=51,52,\cdots 100,
\end{cases}
\label{eq:distribution}
\end{equation}
where $\mu_0=[0,0,0]^\top$, $\mu_1=[0,0,1]^\top$, and $D = [[3,0,0], [0,3,0], [0,0,1]]$. 

Here the means are different at the third dimension, where the lowest variance was imposed among the principal diagonals of $D$ (Supplementary Figure 1).
Therefore, a classical MDS does not distinguish groups in a two-dimensional configuration, but performing a multivariate hypothesis testing on the datasets, e.g., PERMANOVA \cite{anderson01}, indicates there is a statistically significant difference between the groups with $P = 0.005$. 
% In this simulated data, a classical MDS does not distinguish the binary groups in two-dimensional configuration ($p = 0.914$) because the difference is in the third dimension with the lowest variance among the principal diagonals (Figure \ref{fig:mds_sim_data}). 

Using the simulated dataset, we measured the normalized stress (Equation \ref{eq:mds_stress1}) and correlation in Shepard diagram.
Using Algorithm \ref{alg:mm} we confirmed the configuration converged over a number of iterations for a range of hyperparameter up to $\lambda= 0.7$, 
a value that ensured the confirmatory term in Equation \ref{eq:fmds} becomes negligible compared to MDS term (Figure \ref{fig:sim_iter}).

\begin{figure}[h!]
    \centering
    \includegraphics[width=6.5in]{submissions/nips/sim-cmds-iter.pdf}
    \caption[Comparison of MDS and confirmatory terms by iteration in MM algorithm.]{Comparison of MDS and confirmatory terms by iteration of Algorithm \ref{alg:mm} for a range of $\lambda$. Each term values were calculated using simulated dataset following Equation \ref{eq:distribution}.}
    \label{fig:sim_iter}
\end{figure}

Figure \ref{fig:mds_eval_sim} shows a summarized result of the performance of FMDS compared to existing MDS methods.
Regardless of a choice of the hyperparameter $\lambda$, FMDS visualization exhibited a consistent performance with its stress $\sim0.2$ and correlation coefficient $>0.9$ (Figure \ref{fig:mds_eval_sim}a). 
% suggesting either configuration can be used for visualizing the simulated data \cite{kruskal64, borg97b}. 
The behavior is in contrast to the previous supervised MDS \cite{witten11} which was monotonically dependent to $\lambda$, as expected, because SMDS distinguishes groups at the expense of the original distance structure. 
Overall, the stress obtained from the proposed MDS is consistently lower than those from SMDS. 
Similarly, as displayed in Figure \ref{fig:mds_eval_sim}b, Shepard plot shows the proposed FMDS presents a higher correlation of the sample pair distance in between the original and two-dimensional space.

\begin{figure}[h]
    \centering
    \includegraphics[width=5.5in]{submissions/nips/eval_sim.png}
    \caption{Evaluation of proposed FMDS using simulated data. (a) Performance of FMDS compared to supervised MDS \cite{witten11} by measuring stress and Pearson correlation coefficient from Shepard plot. (b) Shepard plot of the proposed MDS comparing to SuperMDS for a hyperparameter $\lambda=0.5$. More Shepard plots are provided in Supplementary Figure 2.}
    \label{fig:mds_eval_sim}
\end{figure}

We then compare the performance of FMDS to other dimensionality reductions such as UMAP \cite{mcInnes18}, that are widely used in visualizing multivariate data. 
While UMAP does not employ the same hyperparameter as FMDS or SMDS, we evaluated its performance by varying the size of local neighborhood, used for manifold approximation, from 5 to 30.
For a better comparison supervised learning mode with UMAP is also implemented.
Table \ref{tab:eval_sim} shows the summarized result of performance by FMDS, SMDS, and UMAP with two different modes for the range of hyperparameters using simulated dataset.
The result suggests the proposed FMDS produces a 2D configuration by imposing a consistently lower distortion of distance structure compared to other methods.

\begin{table}[h]
    \caption{Summary of performance of dimensionality reduction methods using simulated data.}
    \centering
    \begin{tabular}{c|c c c c}
        & FMDS & SMDS \cite{witten11} & UMAP \cite{mcInnes18} & UMAP (supervised) \cite{mcInnes18} \\
        \hline
        Stress-1 & 0.15 -- 0.20 & 0.13 -- 0.42 & 0.32 -- 0.44 & 0.61 -- 0.70 \\
        Pearson correlation & 0.90 -- 0.96 & 0.70 -- 0.96 & 0.69 -- 0.77 & 0.10 -- 0.30 \\
    \end{tabular}
    \label{tab:eval_sim}
\end{table}

\subsubsection{Microbial community}
We next take a microbial community as another dataset to compare the performances of different dimensionality reductions.
The dataset presents a compositional structure expressed by abundance of 16S rRNA gene of 72 bacterial taxa.
As a distance metric, the weighted Unifrac \cite{lozupone07} is chosen to obtain pairwise distance between individual community samples.
We considered two datasets (Site 1 and Site 2) where each contains thirty-six, balanced community samples with a binary label (e.g., with or without a presence of bacterial host) \cite{kim22}. 

The performance of FMDS is evaluated by calculating Stress-1 and the Shepard plot visualization. 
Again we observe that FMDS configuration produced a low stress that is less dependent on the hyperparameter $\lambda$, which even decreases with nonzero $\lambda$ (0.1, 0.3) than a classical MDS (Figure \ref{fig:mds_eval_sites}a). 
Shepard plot and Pearson correlation also show that the configurations nicely preserve the original distance in the microbial community data, except for a case when the largest $\lambda$ = 0.5 is applied to Site 1 community dataset.

\begin{figure}[ht]
    \centering
    \includegraphics[width=4.5in]{submissions/nips/eval_site.png}
    \caption[Evaluation of proposed MDS using community data.]{Evaluation of the FMDS using microbial community data, measured by (a) Stress-1 and Pearson correlation from Shepard plot. (c) Shepard plot from each sample site. More results are provided in Supplementary Figure 4.}
    \label{fig:mds_eval_sites}
\end{figure}

\paragraph{Self-supervised learning with SimCLR and PopPhy-CNN.}
In addition to the above, we also sought to compare our FMDS with existing neural network models that are used for dimensionality reduction.
For converting compositional microbial abundance with its phylogenetic information into a matrix, we implement PopPhy-CNN \cite{reiman20} architecture.
In brief, the encoder consists of one Gaussian noise filter, two 2D convolution layers (with kernel size of 5 by 3), and one fully connected layer with 32 output nodes.
For the self-supervised learning framework we choose SimCLR \cite{chen20}, where the data augmentation is performed by applying random brightness and contrast filter.
In a pretraining step of SimCLR, a model is constructed by compiling encoder, projection head (two dense layers each of 32 output nodes), and one dense layer (10 output nodes).
Site 1 and 2 datasets are merged and split into train and evaluation data (60, 12 each) for the pretraining step, resulting a linear probing accuracy of 53.3\% after 50 training epochs.
In the following finetuning step, the encoder is added with linear probe, which resulted in a validation accuracy of 83.3\% after 50 epochs.
Trained encoder is used to obtain a 32 nodes-sized feature for each microbial community sample in Site 1 and 2.
Detailed parameters and steps for training / evaluating the neural network architecture is described in Supplementary note 3.

\paragraph{Performace evaluation of FMDS with other methods.}
Table \ref{tab:eval_sites} shows the summarized performance of FMDS with microbiome dataset compared to existing dimensionality reduction methods (UMAP, neural network architecture).
Again we observe that the proposed MDS delivers a two-dimensional configuration with less distorted distance structure when compared to self-supervised learning model (SimCLR) or supervised UMAP.
Overall, the evaluation suggests that FMDS outperforms existing (self-) supervised learning tools by preserving the global structure in regards to the pairwise distances.

\begin{table}[h]
    \caption{Summary of dimensionality reduction tool performances using 72 microbiome samples.}
    \centering
    \begin{tabular}{c|c c c c}
        & \begin{tabular}{@{}c@{}}FMDS \\ (proposed)\end{tabular} & \begin{tabular}{@{}c@{}}SimCLR \cite{chen20} + \\ PopPhy-CNN \cite{reiman20}\end{tabular} & UMAP & UMAP (supervised) \\
        \hline
        Stress-1 & 0.21 -- 0.40 & 0.54 & 0.31 -- 0.54 & 0.63 -- 0.69 \\
        Pearson correlation & 0.61 -- 0.92 & 0.076 & 0.63 -- 0.69 & 0.06 -- 0.29 \\
    \end{tabular}
    \label{tab:eval_sites}
\end{table}



\begin{table}[h]
    \caption{Summary of dimensionality reduction tool performances}
    \centering
    \begin{tabular}{c|c c c}
         & Simulation & Site 1 & Site 2 \\
         \hline
         MDS	&0.9589 &	0.8759& 0.9123 \\
         FMDS &0.9281& 0.8321& 0.8657 \\
         UMAP sup &0.3431& 0.2958& 0.0689 \\
         UMAP uns& 0.7702& 0.7285& 0.6812 \\
         t-SNE& 0.6187& 0.5385& 0.5512 \\
         Isomap& 0.9562& 0.8483& 0.8571 \\
         SMDS& 0.5168& 0.8136& 0.5154
    \end{tabular}
    \label{tab:eval_all}
\end{table}



\subsection{Discriminant analysis with FMDS} \label{sec:vis_fmds}
We next demonstrate how FMDS can handle two-dimensional configuration by addressing sample group difference. 
First, we consider the simulated dataset (Section \ref{sec:eval_sim}) again where a classical MDS does not distinguish the binary groups in two-dimensional configuration ($P = 0.914$) because the group difference lies in the third dimension which is not identifiable (Figure \ref{fig:mds_config_sim}a). 
However, the group difference becomes more visible when FMDS is employed, as shown in Figure \ref{fig:mds_config_sim}b,c.
The distinctions are also verified by a low $P$-value resulting from PERMANOVA test using the two-dimensional configurations ($P = 0.003$).

\begin{figure}[h!]
    \centering
    \includegraphics[width=6in]{submissions/nips/config_sim.pdf}
    \caption[2D visualization of simulated data using MDS.]{2D visualization of simulated data using MDS. Configurations are obtained by setting a hyperparameter (a) $\lambda=0$ (classical MDS), (b) $\lambda=0.3$, and (c) $\lambda=0.5$. For each configuration, a $p$-value is given based on PERMANOVA test.}
    \label{fig:mds_config_sim}
\end{figure}

\paragraph{Bacterial community dataset.}
Finally, we demonstrate how FMDS visualization can improve interpreting microbiome structure while addressing $F$-test results at the same time. 
% To do this a bacterial community data is taken as an example where each dataset contains thirty-six, balanced samples of a binary label (e.g., with or without a presence of microbial host) \cite{kim22}. In detail, each data represents expression levels 16S rRNA gene of 72 bacterial taxa, and the distance between samples is measured using the weighted Unifrac \cite{lozupone07}, a non-Euclidean metric that is commonly used in microbial ecology.
Our datasets, Site 1 and 2, present a unique case where a classical two-dimensional MDS does not fully explain statistical test results on group differences. 
As shown in Figure \ref{fig:mds_config_sites}a, groups in site 1 are dispersed in a different location whereas site 2 groups are not, when visualized using the classical MDS. 
In both Sites, however, moderately small $P$-values are obtained ($<0.1$, Table \ref{tab:site_stat}), indicating the group difference in the community structure is, in fact, statistically significant.

We then visualize the microbiome data using classical and FMDS and compare the configurations.
As expected, for Site 1 community samples the configuration retains its distinction between the class labels regardless of the choice of the hyperparameter $\lambda$ (Figure \ref{fig:mds_config_sites}). 
Moreover, for Site 2 samples we observe a higher distinction between the groups with increasing $\lambda$ when compared to classical MDS (Figure \ref{fig:mds_config_sites}b, Supplementary Figure 3). 
The observation with the visualizations is justified by a quantitative measure using $P$-value calculated on the 2D configurations (Table \ref{tab:site_stat}).
% Overall, it exemplifies FMDS reflects the true difference between groups better

\begin{figure}[ht]
    \centering
    \includegraphics[width=6.5in]{submissions/nips/config_site.pdf}
    \caption{Visualization of algal microbiome data using FMDS. (a) Classical and (b) proposed FMDS comparing two-dimensional configurations sampled at Site 1 and 2. More configurations with other value of hyperparameters are displayed in Supplementary Figure 3.}
    \label{fig:mds_config_sites}
\end{figure}

\begin{table}[h]
    \caption{Statistical significance on the group difference between two treatments using PERMANOVA test \cite{anderson01}.}
    \centering
    \begin{tabular}{c|c c}
        $P$-value & Site 1 & Site 2 \\
        \hline
        Original & $<$0.001 & 0.093 \\
        Classical MDS & $<$0.001 & 0.05 \\
        FMDS ($\lambda=0.5$) & $<$0.001 & 0.099 \\
    \end{tabular}
    \label{tab:site_stat}
\end{table}


%%%%%
\section{Conclusion}
In this work, a self-supervised multidimensional scaling is proposed based on the \textit{F}-statistic and is characterized by comparing the configuration to hypothesis testing results. Using simulated dataset and ecological samples, it is shown that the proposed MDS outperforms an existing method for addressing class labels, as evaluated by its stress and Shepard plot. One limitation of FMDS is that it requires a high computational cost compared to existing methods, in its algorithmic performance during iterations (Supplementary note 4).

A special attention has been given on its behavior where its performances were less dependent on the choice of hyperparameter to an extent where algorithm effectively converges. The finding suggests that the new approach mediates the downside of typical confirmatory MDS as a dimensionality reduction tool, in which stress minimization has been underscored by a discriminatory purpose via a high value of hyperparameter setting. In practice, users may avoid the hassle of hyperparameter selection using such as cross-validation.

Taken together, The proposed MDS can be useful for interpreting a biological dataset with more visual information to explain a hypothesis testing result at the same time, thereby presenting a broader applicability of MDS in modern biological and multivariate data analysis.


%%%%
\section*{Acknowledgements}
The work was supported by the Department of Energy’s Genome Sciences Program grant SCW1039. H.K. was partly supported by the Kwanjeong Educational Foundation. S.K. was partly supported by Global Korea Scholarship (GKS).

\begin{singlespace}
\bibliographystyle{acm}
\bibliography{refs}
\end{singlespace}


% \import{./}{supp.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\end{document}