%%
%% Copyright 2022 OXFORD UNIVERSITY PRESS
%%
%% This file is part of the 'oup-authoring-template Bundle'.
%% ---------------------------------------------
%%
%% It may be distributed under the conditions of the LaTeX Project Public
%% License, either version 1.2 of this license or (at your option) any
%% later version.  The latest version of this license is in
%%    http://www.latex-project.org/lppl.txt
%% and version 1.2 or later is part of all distributions of LaTeX
%% version 1999/12/01 or later.
%%
%% The list of all files belonging to the 'oup-authoring-template Bundle' is
%% given in the file `manifest.txt'.
%%
%% Template article for OXFORD UNIVERSITY PRESS's document class `oup-authoring-template'
%% with bibliographic references
%%

%%%CONTEMPORARY%%%
\documentclass[unnumsec,webpdf,contemporary,large]{template}%
%\documentclass[unnumsec,webpdf,contemporary,large,namedate]{oup-authoring-template}% uncomment this line for author year citations and comment the above
%\documentclass[unnumsec,webpdf,contemporary,medium]{oup-authoring-template}
%\documentclass[unnumsec,webpdf,contemporary,small]{oup-authoring-template}

%%%MODERN%%%
%\documentclass[unnumsec,webpdf,modern,large]{oup-authoring-template}
%\documentclass[unnumsec,webpdf,modern,large,namedate]{oup-authoring-template}% uncomment this line for author year citations and comment the above
%\documentclass[unnumsec,webpdf,modern,medium]{oup-authoring-template}
%\documentclass[unnumsec,webpdf,modern,small]{oup-authoring-template}

%%%TRADITIONAL%%%
%\documentclass[unnumsec,webpdf,traditional,large]{oup-authoring-template}
%\documentclass[unnumsec,webpdf,traditional,large,namedate]{oup-authoring-template}% uncomment this line for author year citations and comment the above
%\documentclass[unnumsec,namedate,webpdf,traditional,medium]{oup-authoring-template}
%\documentclass[namedate,webpdf,traditional,small]{oup-authoring-template}

%\onecolumn % for one column layouts

%\usepackage{showframe}

\graphicspath{{Fig/}}

% line numbers
%\usepackage[mathlines, switch]{lineno}
%\usepackage[right]{lineno}
\DeclareMathOperator*{\argmin}{arg\,min}
\DeclareMathOperator*{\argmax}{arg\,max}
\DeclareMathOperator*{\bbr}{\mathbb{R}}
\newcommand{\1}{{\rm 1}\kern-0.24em{\rm I}}
\newcommand{\cN}{\mathcal{N}}

\theoremstyle{thmstyleone}%
\newtheorem{theorem}{Theorem}%  meant for continuous numbers
%%\newtheorem{theorem}{Theorem}[section]% meant for sectionwise numbers
%% optional argument [theorem] produces theorem numbering sequence instead of independent numbers for Proposition
\newtheorem{proposition}[theorem]{Proposition}%
%%\newtheorem{proposition}{Proposition}% to get separate numbers for theorem and proposition etc.
\theoremstyle{thmstyletwo}%
\newtheorem{example}{Example}%
\newtheorem{remark}{Remark}%
\theoremstyle{thmstylethree}%
\newtheorem{definition}{Definition}



\begin{document}

\journaltitle{Submitted to ISMB 2024}
% \DOI{DOI HERE}
% \copyrightyear{2022}
% \pubyear{2019}
% \access{Advance Access Publication Date: Day Month Year}
% \appnotes{Paper}

\firstpage{1}

%\subtitle{Subject Section}

\title[$F$-informed multidimensional scaling]{Multidimensional scaling informed by $F$-ratio: Visualizing microbiome for inference}

\author[1,2,$\ast$,$\dagger$]{Hyungseok Kim}
\author[3,$\ast$]{Soobin Kim}
\author[4]{Megan M. Morris}
\author[4]{Jeffrey A. Kimbrel}
\author[4]{Xavier Mayali}
\author[1,$\ddagger$]{Cullen R. Buie}

\authormark{Kim et al.}

\address[1]{\orgdiv{Department of Mechanical Engineering}, \orgname{Massachusetts Institute of Technology}, \orgaddress{\street{Cambridge}, \state{MA}, \country{USA}}}
\address[2]{\orgdiv{Institute for Data, Systems, and Society}, \orgname{Massachusetts Institute of Technology}, \orgaddress{\street{Cambridge}, \state{MA}, \country{USA}}}
\address[3]{\orgdiv{Department of Statistics}, \orgname{University of California, Davis}, \orgaddress{\street{Davis}, \state{CA}, \country{USA}}}
\address[4]{\orgdiv{Physical and Life Sciences Directorate}, \orgname{Lawrence Livermore National Laboratory}, \orgaddress{\street{Livermore}, \state{CA}, \country{USA}}}

\corresp[$\ast$]{Equal contribution.\vfill}
\corresp[$\dagger$]{Present address: Department of Chemical Engineering, Massachusetts Institute of Technology, Cambridge, MA, USA\vfill}
\corresp[$\ddagger$]{Corresponding author. \href{email:crb@mit.edu}{crb@mit.edu}}

\received{25}{1}{2024}
% \revised{Date}{0}{Year}
% \accepted{Date}{0}{Year}

%\editor{Associate Editor: Name}

%\abstract{
%\textbf{Motivation:} .\\
%\textbf{Results:} .\\
%\textbf{Availability:} .\\
%\textbf{Contact:} \href{name@email.com}{name@email.com}\\
%\textbf{Supplementary information:} Supplementary data are available at \textit{Journal Name}
%online.}

\abstract{
Multidimensional scaling (MDS) is a dimension reduction technique that preserves pairwise distances between observations and has commonly been used for displaying multivariate data including microbiome.
Supervised learning-based MDS alterations have recently been proposed by incorporating labels for training, but their configurations depend on the choice of a hyperparameter, making them susceptible to false positives.
In this study, we present a weakly supervised MDS approach informed by $F$-ratio, the dispersion of observations that share a common binary label.  
We compare our method with existing dimension reduction approaches and show that our visualization accurately represents the $F$-ratio while consistently preserving the global structure.  
Using an algal-associated and human gut microbiome datasets, we show that our new method better illustrates the community's response to the host presence or diseases, suggesting its potential impact on microbiology and ecology data analysis.
}
\keywords{Multidimensional scaling, Microbiome, $F$-statistic, Dimension reduction}


\maketitle

%%%%
\section{Introduction}
Recent advances in microbiome research have empowered multivariate analytics, keeping pace with the growing size and dimensionality of biological data.
% a demand in using appropriate visualization tools for interpreting the data. 
% in the past several decades ...
% For example, the cost of sequencing bacterial samples have been lowered by \textcolor{red}{XX times for the last Y years}, enabling a amore accessible quantification of microbial community compositions. 
Among diverse multivariate datasets, microbiome data are distinguished by their sparse and compositional structure \cite{Gloor17}. 
Moreover, the analysis of microbiome data involves incorporating a context derived from the taxonomy of each microbial component based on its gene information \cite{lozupone05, Martin02, Armstrong22}.

There are two representative statistical approaches in microbiome analysis.
First, the microbiome data is visualized by reducing its structure to a lower dimensional space, a process known as dimensionality reduction.
% By extracting an essential information from the biological data, the dimensionality reduction seeks a visual representation of the multivariate in a lower dimensional space. 
The most conventional approach is multidimensional scaling (MDS), where a configuration is sought in a way that preserves every pairwise distance between the community samples.
% To retain a consistent data structure while performing the dimension reduction, a configuration is sought in a way that preserves dissimilarity between samples, a process known as the multidimensional scaling (MDS). 
% Compared to other methods in dimensionality reduction \cite{tenenbaum00, mcInnes18, maaten08}, MDS retains a global structure in microbiome data by representing a long-range interaction between samples.
Because the microbiome samples are structurally distinguished among multidimensional data (e.g., highly skewed, zero-inflated \cite{gijbels13}), the simplicity of MDS has been preferred over the other methods in dimensionality reduction methods since its inception \cite{Demaine21, Armstrong22}.
On the other hand, recent methods such as UMAP \cite{mcInnes18}, have been used to visualize a high dimensional microbiome dataset by effectively clustering its features associated with their sampling site within the host \cite{Armstrong21}.
% choosing an appropriate distance metric is an important step for performing MDS \cite{lozupone05}.
% In the classical MDS, a configuration is determined in a way that minimizes the difference between the dissimilarity in the original and the lower dimensional space.
% In microbial ecology where the observational data are compositional (i.e., a sample is comprised of different species expressing a level of abundance), the dissimilarity is measured by a difference in expression level of species within samples. 
% For example, when interpreting a microbial community dataset such as 16S rRNA gene expression, a distance metric such as Unifrac \cite{lozupone05, lozupone07} allows to incorporate taxonomic information as well as the compositional structure. 

Another approach in the microbiome analysis is to infer its significance using hypothesis testing or statistical regression. 
Commonly used inference procedures in the field compare multidimensional structures by incorporating data labels, treatments, or explanatory variables.
For example, one way to test for the significance of treatments is by comparing structural dispersion calculated across and within sample groups, such as the $F$-ratio.
In cases where the distribution of the structure is unknown, empirical hypothesis testing is performed using nonparametric statistics \cite{holmes96, gijbels13, clarke93, anderson01}.
Specifically, if the data can be assumed to be independent, the statistic can be obtained by permuting the data labels \cite{holmes96}. 
% Overall, the inference provides a quantitative understanding of the structure.

One of the distinctions between dimension reduction and statistical inference in microbiome analysis lies in the utilization of label information.
Most dimension reduction methods, such as principal component analysis (PCA), MDS, UMAP \cite{mcInnes18}, t-SNE \cite{maaten08}, Isomap \cite{tenenbaum00}, do not necessarily require the data labels in principle.
While recent unsupervised visualization approaches have differentiated microbiome data groups by clustering \cite{Armstrong21}, there still remains a precaution in choosing an appropriate visualization tool. 
This is because an inappropriate dimension reduction can lead to biased results or false positives \cite{mallick17},
considering that designing a microbiome study is challenging due to a stochastic and founder effect \cite{moore16}.
% Therefore, it can be difficult to detect a subtle difference in biological structures when the design is not established.
For example, if sample groups possess a significantly different structure which is not captured through a conventional unsupervised dimension reduction, e.g., samples with a large dispersion, then visualization results may fail to provide an insightful interpretation compared to performing the statistical test.

% Therefore, the quantitative approach is different than most dimension reduction methods since they do not necessarily require the label information.

% Because these two approaches are fundamentally different, linking the results to one other can be obscure depending on data structures.

% data labels are employed in these statistical inferences.
%which does not use label information and is the unsupervised learning approach. 
% This is because most MDS (e.g., PCA, PCoA) are unsupervised learning, whereas the hypothesis testing aims to infer whether responses (e.g., class, label) can influence the distribution of each group of observations. 
% For example, a configuration from the classical MDS is not able to explain a small but statistically meaningful difference between groups of different treatment. 
Indeed, employing labels for visualizing multidimensional data has been accepted in other domain applications.
The motivation underlies by revising the classical approach, such as classical MDS, by including external information that is conferred by group labels \cite{ding18}. 
Broadly termed as the confirmatory MDS \cite{borg97b}, the approach includes applying an external constraint to the MDS, thereby providing a more contextual illustration of the data structure.
These altered MDS configuration are generally accepted up to a point where they do not deviate heavily from the original configuration, as classical MDS can also produce different configurations \cite{borg97a}. 
In the confirmatory MDS methods, an objective function is constructed by adding a label-informed confirmatory term to the MDS objective with a hyperparameter. 
These approaches have been useful in differentiating multivariate data by groups in lower dimensions \cite{cox93, yang18}.
However, the configurations can highly be sensitive to the choice of the hyperparameter. For example, setting an excessive weight on label information may lead to an unbalanced configuration, a phenomenon commonly observed in existing dimension reduction methods. To prevent such potentially misleading outcomes, such as false positives, the hyperparameter is empirically determined through iteration, while this iterative process can be time-limiting, restricting its broader application in biological analysis.

In this work, we introduce a weakly supervised version of MDS that is informed by data labels through $F$-statistic, which we refer to as FMDS.
In other words, FMDS is a visualization method that configures multivariate data structures through their statistical significance. 
Under a binary class setting, the purpose of FMDS is to explain a difference between groups (even if it is small), as an adjustment of the MDS configuration. 
The motivation also sets itself apart from other confirmatory approaches in that FMDS does not directly aim to discriminate between groups. Instead, it focuses on creating configurations that separate groups based on the extent of their true differences.
By characterizing the behavior of the proposed framework we show that FMDS configuration is less dependent on the choice of the model hyperparameter, avoiding the false positiveness.

\subsection{Related Work}\label{sec2}
\subsubsection{Supervised multidimensional scaling}
Consider a balanced design where the number of total observations is $N$, and each observation $\mathbf{x}_i$ is $S$-dimensional, pertaining to a binary label $y_i\in\{0,1\}$ for every $i=1,\cdots N$. 
Using the set of observations $\mathbf{X}=(\mathbf{x}_1,\cdots \mathbf{x}_N)$, a pairwise distance between $\mathbf{x}_i$ and $\mathbf{x}_j$ can be obtained using a metric, e.g., Euclidean, Bray-Curtis \cite{bray57}, UniFrac \cite{lozupone05}. 
Let us define $\mathbf{D} = [d_{ij}]\in[0,\infty)^{N\times N}$ the matrix obtained from the pairwise distance.

In metric multidimensional scaling, a lower-dimensional configuration $\mathbf{Z} = (\mathbf{z}_1, \cdots \mathbf{z}_N)\in \bbr^{N \times 2}$ is sought in a way that preserves the pairwise distance while the dimension is reduced. 
This is enabled by minimizing the ``raw stress'' \cite{borg05c}.
% : ${\rm Stress} = \frac{1}{2}\sum_{i,j} (d_{ij} - \| \mathbf{z}_i - \mathbf{z}_j \|_2 )^2$.
% a difference in the pairwise distance between the dimensions and summing the difference for all pairs, so-called 
The metric MDS is unsupervised learning and it does not require a set of labels $y_i$ for training.

In Supervised multidimensional scaling (SMDS), on the other hand, imposes an additional constraint on the configuration based on class labels \cite{witten11}, with its purpose to discriminate observations by the labels as well as to carry out the classical scaling.
Witten et al. \cite{witten11} proposed an objective function by adding a newly designed confirmatory term to the raw stress, 
\begin{equation}
\begin{split}
O(\mathbf{Z}) 
= &\underbrace{\frac{1}{2}\sum_{i,j} (d_{ij} - \| \mathbf{z}_i - \mathbf{z}_j \|_2 )^2}_\text{raw stress} + \\
&\lambda\underbrace{\sum_{i,j:y_{j}>y_{i}} (y_j - y_i) \sum_{k=1}^2 \left(\frac{d_{ij}}{\sqrt{2}}-(z_{jk} - z_{ik})\right)^2}_\text{confirmatory term},
\end{split}
\end{equation}
where the confirmatory term involves the labels set $[y_i]$. 
Minimizing the objective $O(\mathbf{Z})$ locates the configuration points closer to each other when they are within the same label group.
Additionally, a hyperparameter $\lambda$ balances the confirmatory term with the stress and controls the degree of classification. 
Selection of the hyperparameter is carefully driven by the data structure, so that the process avoids spurious group distinctions during the visualization.

\subsubsection{Non-parametric multivariate hypothesis testing}
Hypothesis testing for a group difference in multivariate data is commonly performed by calculating the $F$-statistic, a ratio comparing inter- and intra-group variances based on pairwise distances.
However, a standard $F$-test requires an assumption that observations are normally distributed with the same variance.
Because the assumption is not usually met for biological data, an alternative approach is proposed to derive a ``pseudo'' $F$-ratio \cite{clarke93}, which is by permuting group labels to create an empirical distribution \cite{anderson01, holmes96}.
To be specific, the pseudo $F$-ratio is defined as
\begin{equation} \label{eq:pseudof}
F = \frac{\sum_{i,j}d_{ij}^2 - 2\sum_{i,j}\1\{y_i=y_j\}d_{ij}^2}{2\sum_{i,j}\1\{y_i=y_j\}d_{ij}^2}\cdot (N-2),
\end{equation}
where $\1\{\cdot\}$ denotes an indicator function. 
While the pseudo $F$ does not necessarily follow the $F$-distribution, an empirical distribution can be constructed instead by permuting the labels for a large enough size of dataset \cite{gijbels13, anderson01}.
In other words, by denoting $F^{\pi_k}$ as a new $F$-ratio that is obtained from a permuted labels set $[y_i]^{\pi_k}$, $p$-value is obtained by repeating the permutation $\pi_k$ for $k=1,\cdots,K$:
\begin{equation}
p = \frac{1}{K}\,\sum_{k=1}^{K}\1\{F^{\pi_k}\geq F\}.
\end{equation}
The procedure is known as the permutational multivariate analysis of variance (PERMANOVA) \cite{anderson01}.


%%%%
\section{Materials and methods}
Our goal is to incorporate the multivariate hypothesis testing result for determining an MDS configuration of the dataset.
To achieve the goal, we take a weakly supervised approach with the following two steps.
First, a two-dimensional configuration is trained by performing an unweighted metric MDS (i.e., unsupervised) to initialize the configuration.
Next, the configuration points are adjusted in a way that best represents a $p$-value calculated from the multivariate, binary-labeled dataset.
% we propose a new approach called the $F$-informed multidimensional scaling (FMDS). 
This is enabled by adding a confirmatory term to the raw stress, a similar approach proposed by the supervised MDS \cite{witten11},
\begin{equation} \label{eq:fmds}
    O(\mathbf{Z}) = 
    \underbrace{\sum_{i,j} (d_{ij} - \| \mathbf{z}_i - \mathbf{z}_j \|_2 )^2}_\text{raw stress} +
    \lambda\cdot \underbrace{\argmin_\mathbf{z}\,\left|F_\mathbf{z} - f_\mathbf{z}(F)\right|}_\text{confirmatory term},
\end{equation}
in which the confirmatory term is designed to minimize a difference in $p$-value based on pseudo $F$'s calculated under the original $S$-dimension and the two-dimension, referred to as $F$ and $F_\mathbf{z}$, respectively. 
Again, each \textit{p}-value is obtained based on the empirical, pseudo $F$-distribution via permutation \cite{anderson01}. 
Noting that a scale of these statistics can vary by the dimension and the hyperparameter, we introduce a regression function $f_\mathbf{z}(F):\bbr\rightarrow\bbr$ that maps $F$-ratio from $S$-dimension onto two-dimension (Supplementary Note 1).

In detail, the regression is carried out by iterating permutations of the label set $[y_i]$ for each dimension to obtain two pseudo $F$'s independently:
\begin{align}
F^{\pi_1} &= 
\left(\frac{\sum_{i,j}d_{ij}^2}{2\sum_{i,j}d_{ij}^2\,\1\{y_i^{\pi_1} = y_j^{\pi_1}\}} -1 \right) \cdot (N-2), \\
F^{\pi_2}_\mathbf{z} &= 
\left(\frac{\sum_{i,j}\|\mathbf{z}_i - \mathbf{z}_j\|_2^2}{2\sum_{i,j}\|\mathbf{z}_i - \mathbf{z}_j\|_2^2\,\1\{y_i^{\pi_1} = y_j^{\pi_2}\}} -1 \right) \cdot (N-2),
\label{eq:map1}
\end{align} 
with the superscript $\pi_j$ denoting the permutation ($j=1,2$).
Note that Equation \ref{eq:map1} represents a pseudo $F$ on a two-dimensional configuration $\mathbf{z}$. 
Using the sequence of pair $(F^{\pi_1}, F_\mathbf{z}^{\pi_2})$, the mapping function $f_\mathbf{z}$ is derived by a local regression. 

In practice, the confirmatory term for our FMDS objective can be derived by seeking a configuration $\mathbf{Z}$:

\begin{align}
&\argmin_\mathbf{Z}\,\left|F_\mathbf{z} - f_\mathbf{z}(F)\right| \\
&\approx \argmin_\mathbf{Z}\,\left|\sum_{i,j} \left[1- 2\1\{y_i = y_j\} \left(1+\frac{f_\mathbf{z}(F)}{N-2}\right)\right] \|\mathbf{z}_i - \mathbf{z}_j\|_2^2\right|,
\end{align}
with a detailed derivation provided in Supplementary Note 1.
Algorithm \ref{alg:mapping} describes a pseudocode that derives the mapping function $f_\mathbf{Z}$.

\begin{algorithm}[h]
\caption{Mapping from $F$ to $F_\mathbf{z}$ with random permutation.}
\label{alg:mapping}
\begin{algorithmic}[1]
\Require{
\begin{tabular}{@{}ll}
    $\mathbf{Z} \in \bbr^{N \times 2}$ & : configuration \\
    $\mathbf{D} \in \bbr_{+}^{N \times N}$ & : pairwise distance \\
    $\mathbf{y} = \{0, 1\}^N$ & : labels
\end{tabular}
}
\Ensure{
\begin{tabular}{@{}ll}
    $f_\mathbf{z}:\bbr\rightarrow\bbr$ & : mapping function    
\end{tabular}
}
\State$\mathcal{L} \Leftarrow$ empty list
\State$\mathcal{L}_\mathbf{z} \Leftarrow$ empty list
\For{$1\leq i \leq 999$}
    \State$\mathbf{y}^{\pi_1} \Leftarrow$ random permutation on $\mathbf{y}$
    \State$\mathbf{y}^{\pi_2} \Leftarrow$ random permutation on $\mathbf{y}$
    \State$F^\pi \Leftarrow$ $F$-ratio using $\mathbf{D}$ and $\mathbf{y}^{\pi_1}$ (Equation S1)
    \State$F^\pi_\mathbf{z} \Leftarrow$ $F$-ratio using $\mathbf{Z}$ and $\mathbf{y}^{\pi_2}$ (Equation S2)
    \State$\mathcal{L} \Leftarrow$ append $F^\pi$ to $\mathcal{L}$
    \State$\mathcal{L}_\mathbf{z} \Leftarrow$ append $F^\pi_\mathbf{z}$ to $\mathcal{L}_\mathbf{z}$
\EndFor
\State$\mathcal{L} \Leftarrow$ sort $\mathcal{L}$ by an increasing order
\State$\mathcal{L}_\mathbf{z} \Leftarrow$ sort $\mathcal{L}_\mathbf{z}$ by an increasing order
\State$f_\mathbf{z} \Leftarrow$ local regression from $\mathcal{L}$ to $\mathcal{L}_\mathbf{z}$
\end{algorithmic}
\end{algorithm}


\subsection{Majorize-Minimization (MM) algorithm}
Minimizing raw stress to perform MDS has been achieved by several iteration methods in the past few decades, part of which have originated in the field of graph drawing \cite{Kamada89}. 
Because the stress is a non-convex function in terms of $\mathbf{Z}$, there is no global minimum and it may require a different initialization \cite{Demaine21, zheng19}.
This has also led with proposing several optimization methods including 2D Newton–Raphson \cite{Kamada89}, stochastic gradient descent \cite{zheng19, Kruskal64}, divide-and-conquer \cite{Yang06, Qu15}, and majorization \cite{deLeeuw88}. 

In our problem setting, we implemented the majorization (or Majorize-Minimization) approach to minimize the FMDS objective (Equation \ref{eq:fmds}) as similarly described by \cite{borg05c} and \cite{witten11}.
% Because the confirmatory term in Equation \ref{eq:fmds} is quadratic in terms of $\mathbf{Z}$, we can analytically derive the Majorize-Minimization (MM) algorithm to minimize $O(\mathbf{z})$ under a mild hyperparameter $\lambda$, a typical approach in MDS optimization task \cite{borg05c}. 
In each step of majorization, an optimal configuration point 
\begin{equation}
    \mathbf{z}_k^* = \argmin_{\mathbf{z}_k}O(\mathbf{Z}).
\end{equation}
is sought for every $k=1,\cdots N$ while other points except for $\mathbf{z}_{k}$ are fixed.
Majorizing with Equation \ref{eq:fmds} results in a quadratic expression in terms of $\mathbf{z}_k$,
which can be locally minimized by taking a derivative with respect to $z_{ks}$ and setting to zero, given that the expression is convex with assumptions.
Detailed procedure is described in Supplementary Note 2 and the resulting update rule is given in Algorithm \ref{alg:mm}. 

\begin{algorithm}[h]
\caption{Majorize-Minimization for FMDS.}
\label{alg:mm}
\begin{algorithmic}[1]
\Require{
\begin{tabular}{@{}ll}
    $\lambda \in \bbr_{+}$ & : hyperparameter \\
    $\mathbf{D} \in \bbr_{+}^{N \times N}$ & : pairwise distance \\
    $\mathbf{y} = \{0, 1\}^N$ & : labels    
\end{tabular}
}
\Ensure{
\begin{tabular}{@{}ll}
    $\mathbf{Z} \in \bbr^{N \times 2}$ & : configuration
\end{tabular}
}
\State$F \Leftarrow$ $F$-ratio using $\mathbf{D}, \mathbf{y}$ (Equation \ref{eq:pseudof})
\For{$1\leq t \leq T$}
\For{$1\leq k \leq N$}
    \State$\delta(\mathbf{Z}) \Leftarrow$ sign of confirmatory term (Equation S13)
    \State$f_\mathbf{z}(F) \Leftarrow$ mapping function (Algorithm \ref{alg:mapping})
    \State$\mathbf{z}_k \Leftarrow$ $\mathbf{z}_k$ using $\delta(\mathbf{z})$ and $f_\mathbf{z}(F)$  (Equation S17)
\EndFor
\EndFor
\end{algorithmic}
\end{algorithm}


%%%%
\section{Results}
In this section, we first characterize the behavior of FMDS across a range of hyperparameters and compare its performance to benchmark dimension reduction methods, including MDS, supervised MDS \cite{witten11}, UMAP \cite{mcInnes18}, t-SNE \cite{maaten08}, Isomap \cite{tenenbaum00} and a self-supervised neural network such as SimCLR \cite{chen20}.
For the evaluation, we consider two types of datasets, simulated and experimental, both of which served as examples where the two-dimensional configurations from classical MDS were not consistent with the hypothesis testing results.
Next, we demonstrate how FMDS can improve visualizing biological samples in a way that multivariate $F$-test results are addressed.

\subsection{Evaluating performance of $F$-informed MDS}

Inspired by the evaluation metric proposed by \cite{rhodes21}, we provide the following numerical procedure to evaluate the performance of each dimension reduction method. First, calculate the permuted $F$-ratio, using the original data while permuting the labels. Second, calculate the permuted $F$-ratio with the configurations. Third, compute the correlation between the two sets of $F$-ratios, namely $F$-correlation ($F$-cor). If the configurations show similar dispersion patterns to the original data with permuted labels, then the $F$-correlation would have high values. Finally, calculate $p$-ratio, the ratio between the permutation-based $p$-value of the original data and the embedding. The $p$-ratio being close to 1 implies that the configuration conveys an accurate inference regarding differences in labels and that the dimension reduction method is insensitive to false positives.

Additionally, we quantify a degree of deviation of the configuration $\mathbf{z}$ from the original distances $d_{ij}$.
It is carried out by calculating two metrics; the first is the ``normalized'' stress or Stress-1 \cite{borg05c},
\begin{equation}
\text{Stress-1} = \frac{\sum_{i,j} (d_{ij} - \| \mathbf{z}_i - \mathbf{z}_j \|_2)^2}{\sum_{i,j} \| \mathbf{z}_i - \mathbf{z}_j \|_2^2},
\label{eq:mds_stress1}
\end{equation}
and the second using a Shepard diagram \cite{dexter18} and its correlation coefficient (Cor). 
 Both terms measure how much distortion from the original global structure has occurred in the configuration from each reduction method.

\subsubsection{Simulated dataset} \label{sec:eval_sim}
We first consider a three-dimensional dataset of two balanced groups that follow normal distributions slightly different in means, $\mu_0$, $\mu_1$, but the same covariance matrix $D$. 
Such set of observations $x_i$ can be constructed as, for example, 
\begin{equation}
x_i \sim 
\begin{cases}
\cN \left(\mu_0, D\right), & i=1,2,\cdots 50 \\
\cN \left(\mu_1, D\right), & i=51,52,\cdots 100,
\end{cases}
\label{eq:distribution}
\end{equation}
where $\mu_0=[0,0,0]^\top$, $\mu_1=[0,0,1]^\top$, and $D = [[3,0,0], [0,3,0], [0,0,1]]$. 

Here the means are different at the third dimension, where the lowest variance was imposed among the principal diagonals of $D$ (Supplementary Figure 1).
Therefore, the MDS does not distinguish groups in a two-dimensional configuration, but performing a multivariate hypothesis testing on the datasets, e.g., PERMANOVA \cite{anderson01}, indicates there is a statistically significant difference between the groups with $p = 0.005$. 
% In this simulated data, a classical MDS does not distinguish the binary groups in two-dimensional configuration ($p = 0.914$) because the difference is in the third dimension with the lowest variance among the principal diagonals (Figure \ref{fig:mds_sim_data}). 

Using the simulated dataset, we measure the normalized stress (Equation \ref{eq:mds_stress1}) and correlation coefficient in the Shepard diagram.
Using Algorithm \ref{alg:mm} we confirmed the configuration converged over a number of iterations for a range of hyperparameters up to $\lambda= 0.7$, 
a value that ensured the confirmatory term in Equation \ref{eq:fmds} becomes negligible compared to the MDS term (Figure \ref{fig:sim_iter}).

\begin{figure}[h!]
    \centering
    \includegraphics[width=3.25in]{submissions/aistats/figures/sim-cmds-iter.pdf}
    \caption[Comparison of MDS and confirmatory terms by iteration in MM algorithm.]{Comparison of MDS and confirmatory terms by iteration of Algorithm \ref{alg:mm} for a range of $\lambda$. Each term values were calculated using simulated dataset following Equation \ref{eq:distribution}.}
    \label{fig:sim_iter}
\end{figure}

We then compare the performance of FMDS to other dimensionality reductions such as MDS, SMDS, UMAP \cite{mcInnes18}, t-SNE \cite{maaten08}, and Isomap \cite{tenenbaum00}, which are widely used in visualizing multivariate data. 
%While UMAP does not employ the same hyperparameter as FMDS or SMDS, we evaluated its performance by varying the size of local neighborhood, used for manifold approximation, from 5 to 30.
For UMAP, the supervised learning mode is also implemented (UMAP-S), as well as the unsupervised version (UMAP-U).
Table \ref{tab:eval_sim} presents a summary of performance results using the simulated dataset, showcasing the optimal outcomes across various hyperparameters for each method.
The result suggests the proposed FMDS produces a 2D configuration with its distance structure less distorted than other methods (as evaluated by low stress and high correlation), as well as carrying a meaningful statistical inference as measured by a high $p$-ratio.

\begin{table}[h]
\caption{Summary of performance of dimension reduction methods using simulated data with four different evaluation metrics, $p$-ratio, $F$-correlation, Stress-1, and Shepard diagram correlation coefficient. \label{tab:eval_sim}}%
\begin{tabular*}{\columnwidth}{@{\extracolsep\fill}lllll@{\extracolsep\fill}}
\toprule
Simulated & {$p$-ratio} & {$F$-cor} & {Stress-1} & {Cor} \\
\midrule
    FMDS	&1 & 0.905	& {0.176}	&0.932\\
    MDS &0.064 &{0.960}	&0.173	&0.959\\
    SMDS	&1.002&0.646	&0.238	&0.875\\
    UMAP-S	&1.002&0.271	&0.820	&0.303\\
    UMAP-U	&0.958&0.776	&0.550	&0.768\\
    t-SNE	&0.998&0.778	&0.975	&0.801\\
    Isomap	&0.016 &0.958	&0.162	&{0.960}\\
\botrule
\end{tabular*}
\end{table}

In addition to its performance evaluation, Figure \ref{fig:mds_eval_sim} shows a detailed characteristic behavior of FMDS compared to supervised MDS.
Regardless of a choice of the hyperparameter $\lambda$, FMDS visualization exhibits a consistent performance with its stress nearly as 0.2 and correlation coefficient higher than 0.9 (Figure \ref{fig:mds_eval_sim}a). 
% suggesting either configuration can be used for visualizing the simulated data \cite{kruskal64, borg97b}. 
The behavior is in contrast to supervised MDS \cite{witten11} which was monotonically dependent to $\lambda$, as expected, because SMDS distinguishes groups at the expense of the original distance structure. 
Overall, the stress obtained from the proposed MDS is consistently lower than those from SMDS. 
Similarly, as displayed in Figure \ref{fig:mds_eval_sim}b, Shepard plot shows the proposed FMDS presents a higher correlation of the sample pair distance in between the original and two-dimensional space.

\begin{figure}[h]
    \centering
    \includegraphics[width=3.25in]{submissions/aistats/figures/eval_sim.pdf}
    \caption{Evaluation of $F$-informed MDS using simulated data. (a) Performance of FMDS compared to supervised MDS \cite{witten11} by measuring stress and Pearson correlation coefficient from Shepard plot. (b) Shepard plot of FMDS compared to supervised MDS for a hyperparameter $\lambda=0.5$. More Shepard plots are provided in Supplementary Figure 2.}
    \label{fig:mds_eval_sim}
\end{figure}

\subsubsection{Algal microbiome}
We next take an algal microbial community as another dataset to compare the performances of different dimension reductions.
The dataset presents a compositional structure expressed by the abundance of 16S rRNA gene of 72 bacterial taxa, samples harvested with proximity to their algal host \cite{kim22}.
As a distance metric, the weighted Unifrac \cite{lozupone07} is chosen to obtain the pairwise distance between individual community samples.
We consider two datasets (Site 1 and Site 2) where each contains thirty-six, balanced community samples with a binary label (e.g., with or without the presence of algal host). 

The performance of FMDS is again evaluated by calculating four metrics, which is then compared to existing dimension reduction methods. 
Table \ref{tab:eval_site1} and Table \ref{tab:eval_site2} show the summarized performance of FMDS with the dataset.
Notably, we observe that the proposed FMDS delivers a two-dimensional configuration with a less distorted distance structure when compared to self-supervised learning neural network (SimCLR, see Supplementary Note 3) or supervised UMAP from in Site 2 algal microbiome.
In addition, compared with unsupervised methods including MDS, unsupervised UMAP, t-SNE, and Isomap, our visualization provides a more accurate inference, as represented by the $p$-ratio being close to unity.
Overall, our evaluation suggests that FMDS outperforms existing dimension reduction tools by preserving the global structure in regard to the pairwise distances, as well as being less prone to false positives.

\begin{table}[h]
\caption{Summary of dimensionality reduction tool performances using Site 1 algal microbiome dataset \label{tab:eval_site1}}%
\begin{tabular*}{\columnwidth}{@{\extracolsep\fill}lllll@{\extracolsep\fill}}
\toprule
Site 1 & {$p$-ratio} & {$F$-cor} & {Stress-1} & {Cor} \\
\midrule
    FMDS &1.000&0.648	  &{0.361}&{0.611} \\
    MDS	 &1.000&{0.925} &0.346 &0.911\\
    SMDS &1.000&	0.880 &0.215&0.909\\
    UMAP-S&1.000&	0.351 &0.994&0.286\\
    UMAP-U &1.000&	0.791 &0.955&0.694\\
    t-SNE &1.000&	0.622&1.000&0.551\\
    Isomap &1.000&0.909& 0.237&0.862\\
    SimCLR &0.481&0.103& 0.990&-0.022\\
\botrule
\end{tabular*}
\end{table}

\begin{table}[h]
\caption{Summary of dimensionality reduction tool performances using Site 2 algal microbiome dataset\label{tab:eval_site2}}%
\begin{tabular*}{\columnwidth}{@{\extracolsep\fill}lllll@{\extracolsep\fill}}
\toprule
Site 2 & {$p$-ratio} & {$F$-cor} & {Stress-1} & {Cor} \\
\midrule
    FMDS	&0.989&0.806	&{0.244}	&{0.857}\\
    MDS	   &0.583 &{0.903}	&0.399	&0.877\\
    SMDS	&1.094&0.596	&0.272	&0.834\\
    UMAP-S	&1.094&0.103	&0.990	&0.069\\
    UMAP-U	&0.138&0.687	&0.964	&0.479\\
    t-SNE	&0.247&0.598	&1.000	&0.593\\
    Isomap	&0.354&0.849	&0.275	&0.828\\
    SimCLR	&1.094&0.057	&0.987	&0.013\\
\botrule
\end{tabular*}
\end{table}

Additionally, the detailed performance of FMDS is characterized by varying the hyperparameter as shown in Figure \ref{fig:mds_eval_sites}.
Again we observe that FMDS configuration produced a low stress that is less dependent on the hyperparameter $\lambda$, which even decreases with nonzero $\lambda$ (0.1, 0.3) than MDS (Figure \ref{fig:mds_eval_sites}a). 
Shepard plot and Pearson correlation also show that the configurations nicely preserve the original distance in the microbial community data, except for a case when the largest $\lambda$ = 0.5 is applied to Site 1 community dataset (Supplementary Figure 4).

\begin{figure}[ht]
    \centering
    \includegraphics[width=3.25in]{submissions/aistats/figures/eval_site.pdf}
    \caption[Evaluation of proposed MDS using community data.]{Evaluation of FMDS using microbial community data, measured by (a) Stress-1 and Pearson correlation from Shepard plot. Shepard plot from each sample site and more results are provided in Supplementary Figures.}
    \label{fig:mds_eval_sites}
\end{figure}

\paragraph{Self-supervised learning with SimCLR and PopPhy-CNN.}
In addition to the traditional dimension reductions, we also sought to compare our FMDS with existing neural network models that are used for dimensionality reduction.
For converting compositional microbial abundance with its phylogenetic information into a matrix, we implemented PopPhy-CNN \cite{reiman20} architecture.
In brief, the encoder consists of one Gaussian noise filter, two 2D convolution layers (with kernel size of 5 by 3), and one fully connected layer with 32 output nodes.
For the self-supervised learning framework we choose SimCLR \cite{chen20}, where the data augmentation is performed by applying random brightness and contrast filter.
In a pretraining step of SimCLR, a model is constructed by compiling encoder, projection head (two dense layers each of 32 output nodes), and one dense layer (10 output nodes).
Site 1 and 2 datasets were individually trained and evaluated (30, 6 data each) for the pretraining step, resulting a linear probing accuracy of 53.3\% after 50 training epochs.
In the following finetuning step, the encoder is added with linear probe, resulting in a validation accuracy higher than 83.3\% after 50 epochs.
The trained encoder is used to obtain a 32 nodes-sized feature for each microbial community sample in Site 1 and 2.
Detailed parameters and steps for training and evaluating the neural network architecture are described in Supplementary Note 3.

\subsubsection{Human microbiome dataset}
In addition to evaluating FMDS using algal microbiome, we further summarize the performance of FMDS by applying it to broader microbiome datasets of larger sizes.
We used two publicly available datasets \cite{Pasolli16, reiman20} that were generated using Shotgun Metagenomic sequencing. For our analysis, we specifically selected datasets that included information at the genus level.
The first dataset is analyzed from a gut microbiome from 118 healthy subjects and 114 liver cirrhosis patients from a single study \cite{Qin14}.
The second microbiome is sampled from a human gut of 223 healthy and 223 patients with type 2 diabetes (T2D) combined by two separate studies \cite{Qin21, Karlsson13}.
In both cases, the PERMANOVA results indicate a significant difference between healthy subjects and patients at a significance level of 0.05, with respective p-values of 0.006 and 0.
As shown in Tables \ref{tab:eval_cirrhosis}, \ref{tab:eval_t2d}, all dimension reductions show a $p$-ratio close to unity, suggesting these data visualizations effectively convey statistical difference observed in the original dimension.
It is notable, however, that FMDS presents the lowest stress among the methods, implying its minimal distortion of the microbiome structure towards its visualization.
Indeed, visualization of the FMDS configuration (Figure \ref{fig:t2d}) suggests that the global data structure is more effectively preserved by uniformly distributing the points, as well as preventing a false positive data interpretation. This is particularly evident when compared to UMAP-S, a supervised dimension reduction method. 

\begin{table}[h]
\caption{Summary of performance of dimension reduction methods using the cirrhosis dataset. \label{tab:eval_cirrhosis}}%
\begin{tabular*}{\columnwidth}{@{\extracolsep\fill}lllll@{\extracolsep\fill}}
\toprule
Cirrhosis & {$p$-ratio} & {$F$-cor} & {Stress-1} & {Cor} \\
\midrule
    FMDS	&	0.998	&	0.707	&	0.325	&	0.782	\\
MDS	&	1.000	&	0.777	&	0.988	&	0.736	\\
UMAP-S	&	1.006	&	0.105	&	0.968	&	0.133	\\
UMAP-U	&	0.978	&	0.563	&	0.907	&	0.503	\\
t-SNE	&	0.990	&	0.411	&	0.995	&	0.559	\\
Isomap	&	1.004	&	0.671	&	0.421	&	0.658	\\
\botrule
\end{tabular*}
\end{table}


\begin{table}[h]
\caption{Summary of performance of dimension reduction methods using the T2D dataset. 
\label{tab:eval_t2d}}%
\begin{tabular*}{\columnwidth}{@{\extracolsep\fill}lllll@{\extracolsep\fill}}
\toprule
T2D & {$p$-ratio} & {$F$-cor} & {Stress-1} & {Cor} \\
\midrule
FMDS	&	1.000	&	0.677	&	0.332	&	0.768	\\
MDS	&	0.998	&	0.727	&	1.040	&	0.716	\\
UMAP-S	&	1.000	&	0.156	&	0.976	&	0.218	\\
UMAP-U	&	0.998	&	0.554	&	0.917	&	0.520	\\
t-SNE	&	1.000	&	0.579	&	0.995	&	0.566	\\
Isomap	&	0.990	&	0.638	&	0.498	&	0.579	\\
\botrule
\end{tabular*}
\end{table}


\begin{figure}[ht]
    \centering
    \includegraphics[width=3.25in]{submissions/ISMB/figures/t2d.pdf}
    \caption{Visualization of T2D patients and healthy human gut microbiome. The following hyperparameters were used: $\lambda=0.3$ (FMDS), neighbors $=5$ (UMAP-S), neighbors $=30$ (UMAP-U), perplexity$=10$ (t-SNE), $K=7$ (Isomap).}
    \label{fig:t2d}
\end{figure}


\subsection{Discriminant analysis with FMDS} \label{sec:vis_fmds}
We next demonstrate how FMDS can handle two-dimensional configuration by addressing sample group differences. 
First, we consider the simulated dataset (Section \ref{sec:eval_sim}) again where a classical MDS does not distinguish the binary groups in two-dimensional configuration ($p = 0.914$) because the group difference lies in the third dimension which is not identifiable (Figure \ref{fig:config_sim}a, Supplementary Figure 1). 
However, the group difference becomes more visible when FMDS is employed, as shown in Figure \ref{fig:config_sim}b.
The distinctions are also verified by a low $p$-value resulting from the PERMANOVA test using the two-dimensional configurations.

\begin{figure}[h!]
    \centering
    \includegraphics[width=3.1in]{submissions/aistats/figures/config_sim.pdf}
    \caption[2D visualization of simulated data using MDS.]{Two-dimensional visualization of simulated data. Configurations are obtained by training with MDS (left) or FMDS with $\lambda=0.7$ (right). For each configuration, an error ellipse (68\% confidence) and $p$-value are presented and compared to PERMANOVA result \cite{anderson01}.}
    \label{fig:config_sim}
\end{figure}

\subsubsection{Algal microbiome}
Finally, we demonstrate how FMDS visualization can improve interpreting microbiome structure while addressing $F$-test results at the same time. 
% To do this a bacterial community data is taken as an example where each dataset contains thirty-six, balanced samples of a binary label (e.g., with or without a presence of microbial host) \cite{kim22}. In detail, each data represents expression levels 16S rRNA gene of 72 bacterial taxa, and the distance between samples is measured using the weighted Unifrac \cite{lozupone07}, a non-Euclidean metric that is commonly used in microbial ecology.
Our datasets, Site 1 and 2, present a unique case where a two-dimensional MDS does not fully explain statistical test results on group differences. 
As shown in Figure \ref{fig:mds_config_sites}a, groups in Site 1 are dispersed in a different location whereas Site 2 groups are not, when visualized using MDS. 
In both sites, however, moderately small $p$-values are obtained from multivariate hypothesis testing ($<0.1$, Table \ref{tab:site_stat}), indicating the group difference in the community structure is, in fact, statistically significant.

We then visualize the microbiome data using MDS and FMDS and compare the configurations.
Indeed, for Site 1 community samples the configuration retains its distinction between the class labels regardless of the choice of the hyperparameter $\lambda$ (Figure \ref{fig:mds_config_sites}a). 
Moreover, for Site 2 samples a higher distinction is observed between the groups with higher $\lambda$ when compared to MDS (Figure \ref{fig:mds_config_sites}b, Supplementary Figure 3). 
The observation with the visualizations is justified by a quantitative measure using $p$-value calculated on the 2D configurations (Table \ref{tab:site_stat}).
% Overall, it exemplifies FMDS reflects the true difference between groups better

\begin{figure}[ht]
    \centering
    \includegraphics[width=3.25in]{submissions/aistats/figures/config_site.pdf}
    \caption{Visualization of algal microbiome data using FMDS. (a) Classical and (b) proposed FMDS comparing two-dimensional configurations sampled at Site 1 and 2. More configurations with other values of hyperparameters are displayed in Supplementary Figure.}
    \label{fig:mds_config_sites}
\end{figure}



\begin{table}[h]
\caption{Statistical significance on the group difference between two treatments using PERMANOVA test \cite{anderson01}\label{tab:site_stat}}%
\begin{tabular*}{\columnwidth}{@{\extracolsep\fill}lll@{\extracolsep\fill}}
\toprule
{$p$-value} & {Site 1} & {Site 2} \\
\midrule
    Original data & $<$0.001 & 0.093 \\
    MDS & $<$0.001 & 0.05 \\
    FMDS ($\lambda=0.5$) & $<$0.001 & 0.099 \\
\botrule
\end{tabular*}
\end{table}



%%%%%
\section{Discussion}
In this work, a weakly supervised multidimensional scaling is proposed based on the $F$-ratio and is characterized by comparing the configuration to hypothesis testing results. Using the simulated datasets and ecological samples, our results demonstrate that the proposed FMDS outperforms existing methods for addressing class labels, as evaluated by various measures. However, one limitation of FMDS is its computationally intensive nature, particularly in its algorithmic performance during iterations (see Supplementary Material).

Special attention has been given to its behavior, showcasing less dependency on the choice of hyperparameter to an extent where the algorithm effectively converges. The finding suggests that the new approach mediates the downside of typical confirmatory MDS as a dimensionality reduction tool, in which stress minimization has been underscored by a discriminatory purpose via heavier weight imposed by the hyperparameter choice. In practical terms, users may alleviate the burden of hyperparameter selection through techniques such as cross-validation.

Taken together, the proposed FMDS proves to be useful for visualizing microbiome datasets at a higher consistency delivering hypothesis testing results. The method presents a broader applicability of MDS in the contemporary biological and multivariate data analysis.

\section{Competing interests}
No competing interest is declared.

% \section{Author contributions statement}
% Must include all authors, identified by initials, for example:
% S.R. and D.A. conceived the experiment(s),  S.R. conducted the experiment(s), S.R. and D.A. analysed the results.  S.R. and D.A. wrote and reviewed the manuscript.

\section{Acknowledgments}
The work was supported by the Department of Energy’s Genome Sciences Program grant SCW1039. H.K. was partly supported by the Kwanjeong Educational Foundation. S.K. was partly supported by Global Korea Scholarship (GKS). The authors thank the anonymous reviewers for their valuable suggestions. This work is supported in part by funds from the National Science Foundation (NSF: \# 1636933 and \# 1920920).


\bibliographystyle{plain}
\bibliography{submissions/ISMB/refs}

\end{document}
